packages/crumbs-vue/components/VeeLoaders/VeeProgressCircular/VeeProgressCircular.vue

```diff
@@ -0,0 +1,264 @@
+  },
+
+  destroyed() {
+    this.intersectionObserver?.disconnect();
+  },
+
+  computed: {
+    veeCalculatedSize(): number {
+      return Number(this.size);
+    },
+
+    veeCircumference(): number {
+      return this.radius * 2 * Math.PI;
+    },
+
+    veeNormalizedValue(): number {
+      return this.indeterminate
+        ? 0
+        : Math.min(100, Math.max(0, Number(this.value)));
+    },
+
+    veeStrokeWidth(): number {
+      return (Number(this.width) / +this.size) * this.veeViewBoxSize * 2;
+    },
+
+    veeStrokeDashOffset(): number {
+      return ((100 - this.veeNormalizedValue) / 100) * this.veeCircumference;
+    },
+
+    veeColor(): string {
+      return this.$crumbs.themesManager.getColorRef(this.color);
+    },
+
+    veeColorUnderlay(): string {
+      return this.$crumbs.themesManager.getColorRef('greyLighten3');
+    },
+
+    veeViewBoxSize(): number {
+      return this.radius / (1 - Number(this.width) / +this.size);
+    },
+
+    veeStyle(): Record<string, any> {
+      return {
+        width: `${this.veeCalculatedSize}px`,
+        height: `${this.veeCalculatedSize}px`,
+        ...this.style,
+      };
+    },
+
+    veeSvgStyle(): Record<string, any> {
+      return {
+        transform: `rotate(${Number(ROTATE)}deg)`,
+      };
+    },
+
+    veeClass(): string {
+      return classBuilder({
+        'vee-progress-circular': true,
+        'vee-progress-circular-indeterminate':
+          this.indeterminate && this.visible,
+        'vee-progress-circular-indeterminate-paused':
+          this.indeterminate && !this.visible,
+      });
+    },
+
+    veeCircleClass(): string {
+      return classBuilder({
+        'vee-progress-circular-overlay': true,
+        'vee-progress-circular-overlay-indeterminate': this.indeterminate,
+      });
+    },
+  },
+});
+</script>
+
+<style scoped>
+.vee-progress-circular {
+  display: inline-block;
+  position: relative;
+  vertical-align: middle;
+  justify-content: center;
+  align-items: center;
+}
+
+.vee-progress-circular svg {
+  width: 100%;
+  height: 100%;
+  margin: auto;
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  z-index: 0;
+}
+
+.vee-progress-circular-indeterminate svg {
+  animation: vee-progress-circular-rotate 1.4s linear infinite;
+  transform-origin: center center;
+  transition: all 0.2s ease-in-out;
+}
+
+.vee-progress-circular-indeterminate-paused {
+  animation-play-state: paused !important;
+}
+
+.vee-progress-circular-content {
+  align-items: center;
+  display: flex;
+  justify-content: center;
+}
+
+.vee-progress-circular-underlay {
+  color: v-bind('veeColorUnderlay');
+  stroke: currentColor;
+  z-index: 1;
+}
+
+.vee-progress-circular-overlay {
+  color: v-bind('veeColor');
+  stroke: currentColor;
+  transition: all 0.6s ease-in-out;
+  z-index: 2;
+}
+
+.vee-progress-circular-overlay-indeterminate {
+  animation: vee-progress-circular-dash 1.4s linear infinite;
+  stroke-linecap: round;
```

Comment:
I'd leave the overlay style the same regardless of `indeterminate` value.. (maybe a round prop too?)